File: src/models/assignment.ts
import type { Attachment } from "./attachment";
import type { AssignmentDifficulty } from "./assignment-difficulty";
import type { AssignmentReturn } from "./assignment-return";
import type { AssignmentTheme } from "./assignment-theme";
import type { Subject } from "./subject";

export type Assignment = Readonly<{
  id: string;
  subject: Subject;
  description: string;
  backgroundColor: string;
  done: boolean;
  deadline: Date;
  attachments: Array<Attachment>;
  difficulty: AssignmentDifficulty;
  /** Time that should take, in minutes, to do the homework. */
  length?: number;
  /** Themes associated with this homework. */
  themes: AssignmentTheme[];
  return: AssignmentReturn
  /**
   * If defined, can be used to retrieve
   * the lesson contents from the resources tab.
   */
  lessonResourceID?: string;
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/Assignment.kt
package ink.literate.pawnote.models

import kotlinx.datetime.LocalDateTime

data class Assignment(
    val id: String,
    val subject: Subject,
    val description: String,
    val backgroundColor: String,
    val done: Boolean,
    val deadline: LocalDateTime,
    val attachments: List<Attachment>,
    val difficulty: AssignmentDifficulty,
    /** Time that should take, in minutes, to do the homework. */
    val length: Double? = null,
    /** Themes associated with this homework. */
    val themes: List<AssignmentTheme>,
    val `return`: AssignmentReturn,
    /** If defined, can be used to retrieve the lesson contents from the resources tab. */
    val resourceID: String? = null
)
